src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -3087,11 +3111,12 @@ internal Task WriteByteArray(byte[] b, int len, int offsetBuffer, bool canAccumu
         }
 
         // This is in its own method to avoid always allocating the lambda in WriteByteArray
-        private void WriteByteArraySetupContinuation(byte[] b, int len, TaskCompletionSource<object> completion, int offset, Task packetTask)
+        private void WriteByteArraySetupContinuation(byte[] array, int len, TaskCompletionSource<object> completion, int offset, Task packetTask)
         {
             AsyncHelper.ContinueTask(packetTask, completion,
-                () => WriteByteArray(b, len: len, offsetBuffer: offset, canAccumulate: false, completion: completion),
-                connectionToDoom: _parser.Connection);
+                () => WriteByteArray(ReadOnlySpan<byte>.Empty, len: len, offsetBuffer: offset, canAccumulate: false, completion: completion, array),
```

Comment:
Can't the continuation be done with a Span instead of passing in the array? If we can operate on Span all through, that seems more consistent.

Reply:
It would be, and that was my original intent. Unfortunately you can't put a span into a heap allocated closure. That's what all the fiddling around with the array is for.

Reply:
Yeah makes sense.